src/unit_tests/test_kernels/result_reuse_kernel.py

```diff
@@ -0,0 +1,51 @@
+import torch
+import torch.nn as nn
+from torch.utils.cpp_extension import load_inline
+
+####
+# 
+# This adversarial kernel utilizes zeros_like which can get allocated the same physical memory as the PyTorch reference outputs
```

Comment:
we actually tested using empty_like which doesn't clear physical memory (zeros) right? only empty_like could cause this kind of issue (zero_like should clear all the memory?)